component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java

```diff
@@ -64,7 +75,31 @@ public boolean validate() throws MetricReportingException {
 
     @Override
     protected Map<String, Object> buildEvent() {
-        eventMap.put(Constants.EVENT_TYPE, "response");
+        eventMap.put(Constants.EVENT_TYPE, Constants.RESPONSE_EVENT_TYPE);
+        // userAgent raw string is not required and removing
+        String userAgentHeader = (String) eventMap.remove(Constants.USER_AGENT_HEADER);
+        // userAgentHeader will not null since it is already validated
+        setUserAgentProperties(userAgentHeader);
         return eventMap;
     }
+
+    private void setUserAgentProperties(String userAgentHeader) {
+        if (uaParser == null) {
+            // if ua parser is not initialised, skip setting properties
+            return;
+        }
+        Client client = uaParser.parse(userAgentHeader);
+        String browser = client.userAgent.family;
+        if (browser == null || browser.isEmpty()) {
+            browser = Constants.UNKNOWN_VALUE;
```

Comment:
We have to set the unknown value if the parser is null too

Reply:
Fixed